tendrl/ceph_integration/manager/__init__.py

```diff
@@ -0,0 +1,138 @@
+import gevent.event
+import logging
+import signal
+
+from tendrl.ceph_integration import ceph
+from tendrl.ceph_integration.manager.cluster_monitor import ClusterMonitor
+from tendrl.ceph_integration.manager import utils
+from tendrl.ceph_integration.central_store import CephIntegrationEtcdPersister
+from tendrl.commons.manager import Manager
+from tendrl.commons.sds_sync import SdsSyncThread
+
+
+LOG = logging.getLogger(__name__)
+
+
+class CephIntegrationSyncStateThread(SdsSyncThread):
+
+    def __init__(self, manager):
+        super(CephIntegrationSyncStateThread, self).__init__()
+
+        self._manager = manager
+        self._complete = gevent.event.Event()
+
+    def _run(self):
+        LOG.info("%s running" % self.__class__.__name__)
+
+        while not self._complete.is_set():
+            data = ceph.heartbeat(heartbeat_type="cluster", discover=True)
+            if data is not None:
+                for tag, cluster_data in data.iteritems():
+                    try:
+                        if tag.startswith("ceph/cluster/"):
+                            if not cluster_data[
+                                    'fsid'
+                            ] in self._manager.clusters:
```

Comment:
this self._manager.on_pull has to go completely, refactor to match node-agent's on pull

https://github.com/Tendrl/node-agent/blob/master/tendrl/node_agent/node_sync/__init__.py#L13